app/src/main/java/de/danoeh/antennapod/dialog/EpisodesApplyActionFragment.java

```diff
@@ -118,62 +164,93 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
-            btnMarkAsUnplayed.setOnClickListener(v -> markedCheckedUnplayed());
-            lastVisibleDiv = R.id.divider3;
-        } else {
-            btnMarkAsUnplayed.setVisibility(View.GONE);
-            view.findViewById(R.id.divider3).setVisibility(View.GONE);
-        }
-        btnDownload = view.findViewById(R.id.btnDownload);
-        if((actions & ACTION_DOWNLOAD) != 0) {
-            btnDownload.setOnClickListener(v -> downloadChecked());
-            lastVisibleDiv = R.id.divider4;
-        } else {
-            btnDownload.setVisibility(View.GONE);
-            view.findViewById(R.id.divider4).setVisibility(View.GONE);
-        }
-        btnDelete = view.findViewById(R.id.btnDelete);
-        if((actions & ACTION_REMOVE) != 0) {
-            btnDelete.setOnClickListener(v -> deleteChecked());
-        } else {
-            btnDelete.setVisibility(View.GONE);
-            if(lastVisibleDiv > 0) {
-                view.findViewById(lastVisibleDiv).setVisibility(View.GONE);
+        saveActionBarTitle(); // needed when we dynamically change the title based on selection
+
+        // Init action UI (via a FAB Speed Dial)
+        mSpeedDialView = view.findViewById(R.id.fabSD);
+        mSpeedDialView.inflate(R.menu.episodes_apply_action_speeddial);
+
+        // show only specified actions, and bind speed dial UIs to the actual logic
+        for (ActionBinding binding : actionBindings) {
+            if ((actions & binding.flag) == 0) {
+                mSpeedDialView.removeActionItemById(binding.actionItemId);
             }
         }
 
+        mSpeedDialView.setOnActionSelectedListener(actionItem -> {
+            ActionBinding selectedBinding = null;
+            for (ActionBinding binding : actionBindings) {
+                if (actionItem.getId() == binding.actionItemId) {
+                    selectedBinding = binding;
+                    break;
+                }
+            }
+            if (selectedBinding != null) {
+                selectedBinding.action.run();
+            } else {
+                Log.e(TAG, "Unrecognized speed dial action item. Do nothing. id=" + actionItem.getId());
+            }
+            return true;
+        });
+
+        showSpeedDialIfAnyChecked();
+
         return view;
     }
 
+    @Override
+    public void onStart() {
+        super.onStart();
+        compatEnsureSpeedDialClickable();
+    }
+
+    @Override
+    public void onStop() {
+        restoreActionBarTitle(); // it might have been changed to "N selected". Restore original.
+        super.onStop();
+    }
+
+    private void compatEnsureSpeedDialClickable() {
+        // On pre-Lollipop devices (that does not support elevation),
+        // need to explicitly bring the fab to the front, otherwise it won't be clickable.
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP) {
+            ViewGroup root = (ViewGroup)getView();
+            root.bringChildToFront(root.findViewById(R.id.fabSDScrollCtr));
```

Comment:
Can't you put it to the front by changing the order of xml elements?

Reply:
I didn't know one could do that, and will get back to you.

Reply:
The issue is now fixed by moving the FAB to the bottom of the xml. Thanks.